### PR TITLE
kube env variables has lower priority than configuration

### PR DESCRIPTION
**Pull request by legionus**
Fixes https://github.com/openshift/image-registry/issues/58

### CODE REVIEW

File: pkg/dockerregistry/server/configuration/configuration_test.go [hunk 1]
```diff
@@ -583,19 +581,79 @@ openshift:
...
 		{
 			setenv: []env{
 				{name: "DOCKER_REGISTRY_SERVICE_HOST", value: "DOCKER_REGISTRY_SERVICE_HOST"},
 				{name: "DOCKER_REGISTRY_SERVICE_PORT", value: "DOCKER_REGISTRY_SERVICE_PORT"},
 			},
-			expected: "DOCKER_REGISTRY_SERVICE_HOST:DOCKER_REGISTRY_SERVICE_PORT",
+			expected: "from-config",
+		},
+		{
+			setenv:   []env{{name: "REGISTRY_OPENSHIFT_SERVER_ADDR", value: "REGISTRY_OPENSHIFT_SERVER_ADDR"}},
```
Comment (bparees) [inline pkg/dockerregistry/server/configuration/configuration_test.go]: this test would be more thorough if it also set the DOCKER_REGISTRY_SERVICE_XXX variables (to ensure they are ignored).

Comment (legionus) [reply pkg/dockerregistry/server/configuration/configuration_test.go]: @bparees I'm not following. These variables will be set on the previous iteration. Each case in this test adds variables, but does not unset them.

What do you mean ?

Comment (bparees) [reply pkg/dockerregistry/server/configuration/configuration_test.go]: ah, I did not realize that, I assumed each TC was independent.

**Pull request merged**